packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -197,6 +197,16 @@ export const App = ({
         return;
       }
 
+      // If we've already successfully initialized the session, don't reinitialize.
+      // This fixes a case where isSignedIn changes from 'true' to 'false', which happens
+      // when a user signs into the browser without a password, and then tries to sign into
+      // Sync later with 2FA. Since entering a password (required for scoped keys) grants
+      // a new sessionToken and the user needs to enter 2FA again for it, we do NOT
+      // want to use the session token from the browser for this case.
+      if (isSignedIn === true) {
```

Comment:
I think this is fine, however it is possible that it might have side effects in other code paths. Our session and cache handline needs some love since its easy to introduce unexpected bugs.

Reply:
Agreed. :/ This has been tweaked now at least and I think is better. I can let (FxA QA, not the third party auth Fx QA) know about this change as well.

Reply:
Another update here, I ended up not needing an `App/index.tsx` change at all. Simply changing `isValid` to check that the user has a valid token instead of a fully verified token worked. This seems like what we'd want anyway.